reo/src/techs.py

```diff
@@ -303,3 +308,178 @@ def default_fuel_burn_rate(size_kw):
             m = 0.0657
             b = 0.004
         return m, b
+
+
+class CHP(Tech):
+    """
+    Includes calcs for converting user-input electric efficiency and thermal recovery fraction to coefficients
+    useable for Xpress (e.g. fuel burn rate in MMBtu/hr/kW_rated (y-intercept)
+
+    """
+
+    def __init__(self, dfm, run_uuid, existing_boiler_production_type_steam_or_hw, oa_temp_degF, site_elevation_ft,
+                 time_steps_per_hour=1, **kwargs):
+        super(CHP, self).__init__()
+
+        self.prime_mover = kwargs.get('prime_mover')
+        self.existing_boiler_production_type_steam_or_hw = existing_boiler_production_type_steam_or_hw
+        self.reopt_class = 'CHP'
+        self.is_chp = True
+        self.time_steps_per_hour = time_steps_per_hour
+        self.derate = 1  # Need to rectify this legacy derate, maybe remove this and replace if no needed (NM/IL?)
+        self.loads_served = ['retail', 'wholesale', 'export', 'storage', 'boiler', 'tes']
+        self.incentives = Incentives(**kwargs)
+
+        try:
+            self.min_kw = kwargs['min_kw']
+            self.max_kw = kwargs['max_kw']
+            self.installed_cost_us_dollars_per_kw = kwargs['installed_cost_us_dollars_per_kw']
+            self.om_cost_us_dollars_per_kw = kwargs['om_cost_us_dollars_per_kw']
+            self.om_cost_us_dollars_per_kwh = kwargs['om_cost_us_dollars_per_kwh']
+            self.elec_effic_full_load = kwargs['elec_effic_full_load']
+            self.elec_effic_half_load = kwargs['elec_effic_half_load']
+            self.thermal_effic_full_load = kwargs['thermal_effic_full_load']
+            self.thermal_effic_half_load = kwargs['thermal_effic_half_load']
+            self.min_turn_down_pct = kwargs['min_turn_down_pct']
+            self.min_allowable_kw = kwargs['min_allowable_kw']
+            self.use_default_derate = kwargs['use_default_derate']
+            self.max_derate_factor = kwargs['max_derate_factor']
+            self.derate_start_temp_degF = kwargs['derate_start_temp_degF']
+            self.derate_slope_pct_per_degF = kwargs['derate_slope_pct_per_degF']
+        except:
+            print("Prime mover type was not specified and not all cost/performance parameters were input "
```

Comment:
API users will never see this print statement. The validation of inputs should be handled in validators.py

Reply:
@zolanaj I will fix this in my branch chp_size_class_cp.